src/world/biome.cpp

```diff
@@ -381,17 +381,12 @@ Biome::init_materials_(const Json::Value& material_data) {
     return out;
 }
 
-std::map<ColorInt, std::pair<const Material*, ColorId>>
+std::unordered_map<ColorInt, MaterialColor>
 Biome::get_colors_inverse_map() const {
-    std::map<ColorInt, std::pair<const Material*, ColorId>> materials_inverse;
+    std::unordered_map<ColorInt, MaterialColor> materials_inverse;
     for (const auto& element : materials_) {
         for (ColorId color_id = 0; color_id < element.second.color.size(); color_id++) {
-            materials_inverse.insert(
-                std::map<ColorInt, std::pair<const Material*, ColorId>>::value_type(
-                    element.second.color.at(color_id).second,
-                    std::make_pair(&element.second, color_id)
-                )
-            );
+            materials_inverse.emplace(color_id, MaterialColor(element.second, color_id));
```

Comment:
```suggestion
            materials_inverse.emplace(color_id, MaterialColor{element.second, color_id});
```

Reply:
```sh
/home/alem/Documents/code_porjects/Fun-Game/src/world/biome.cpp:390:65: error: no matching function for call to ‘terrain::MaterialColor::MaterialColor(<brace-enclosed initializer list>)’
  390 |                 color_id, MaterialColor{element.second, color_id}
      |                                                                 ^
```

Reply:
You need to delete the other ctors from the struct